ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/TransactionImporterImple.java

```diff
@@ -171,7 +172,12 @@ public void removeImportedTransaction (Xid xid) throws XAException
 	 * @param timeout
 	 * @return
 	 */
-	private TransactionImportResult addImportedTransaction(TransactionImple recoveredTransaction, Xid mapKey, Xid xid, int timeout) {
+	private TransactionImportResult addImportedTransaction(TransactionImple recoveredTransaction, Xid mapKey, Xid xid, int timeout) throws XAException {
+		if (!TxControl.isEnabled()) {
+			// TODO needs looking at for the right code and so on and maybe it's different for recover than it is when importing a new transaction
```

Comment:
Is it OK to leave this TODO?

Reply:
If you don't mind, I would leave this TODO here for later, when the Narayana team agrees that the new PoC is good enough to be developed upon.

Reply:
If you need to leave it here, I think you need to add a blocking JBTM for https://issues.redhat.com/browse/JBTM-3893 and cite it here

Reply:
Added JBTM-3893 to this comment. Also, I added this class in JBTM-3893's list of classes to look at. Thanks Tom

Reply:
Is it possible to consider removing the changes to this from this pull request as it is expected to be implemented in https://issues.redhat.com/browse/JBTM-3893 instead?